### PR TITLE
Fix local minimum deconvolution 

### PR DESCRIPTION
**Pull request by robinschmid**
Chromatographic threshold needed number of all scans of a chromatogram to calc quantile
(still need to merge master)

### CODE REVIEW

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_chromatogramdeconvolution/ResolvingUtil.java [hunk 1]
```diff
@@ -157,4 +160,27 @@
...
+  /**
+   * All scans that were used to build a chromatogram or the maximum number of mobility scans in all
+   * Frames used to construct the mobilogram
+   *
+   * @param data
+   * @param dimension
+   * @param selectedScans
+   * @return
+   */
+  public static int selectedTotalScans(@Nonnull IonTimeSeries<? extends Scan> data,
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/featdet_chromatogramdeconvolution/ResolvingUtil.java:172]: Should be called getTotalNumberOfScansInDimension to be more descriptive

Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/dataprocessing/featdet_chromatogramdeconvolution/ResolvingUtil.java]: done

**Comment by robinschmid** [comment]
Merged into master

**Comment by robinschmid** [comment]
There still remains the question what kind of data the other deconvolution modules need.
Currently we only supply the detected data points. I think we should split the modules more and leave the data retrieval to each module.
### Local minimum search needs:
- only data points where the signal was detected in the scan
- insert zeros in holes between two signals?
- the total number of scans to determine the chromatographic threshold (quantiles)

### Noise amplitude deconvolution needs:
- only data points where signal was detected (all non zero data points)

### Other algorithms need:
- Other algorithms might need the full chromatogram with zeros at scans with missing signals

**Comment by SteffenHeu** [comment]
I agree. However, since this fixes some stuff, we should merge this now and keep that in mind. Add it to the Project maybe?

**Comment by SteffenHeu** [comment]
Furthermore, keep in mind that it was implemented this way to be able to resolve the mobility and rt dimension with the same module.

**Comment by robinschmid** [comment]
We definitely need to keep it in mind. mobility and rt should be no problem.

**Pull request merged**